vm/systemSmartContracts/auction.go

```diff
@@ -260,6 +291,24 @@ func (s *stakingAuctionSC) updateStakeValue(registrationData *AuctionData, calle
 	return vmcommon.Ok
 }
 
+func (s *stakingAuctionSC) getVerifiedBLSKeysFromArgs(args [][]byte) [][]byte {
+	blsKeys := make([][]byte, 0)
+	maxNodesToRun := big.NewInt(0).SetBytes(args[0]).Uint64()
+
+	for i := uint64(1); i < maxNodesToRun*2+1; i += 2 {
+		blsKey := args[i]
+		signedMessage := args[i+1]
+		err := s.sigVerifier.Verify([]byte("stake"), signedMessage, blsKey)
```

Comment:
We should use as message the public key used to sign the transaction. Otherwise there are still open attack vectors: suppose that this transaction is not included immediately in a block and an attacker can use the (signedMessage, blsKey) to assign that particular node to its own reward address. Thus the chances are small, they exist.

Reply:
not really an attack. the attacked would risk a lot by doing this, and honest validators would earn more, by attacking themselves. added to verify the pubkey.